streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -90,24 +88,38 @@ public void before() {
         streams = new KafkaStreams(builder.build(), props);
     }
 
+    @After
+    public void cleanup() {
+        if (streams != null) {
+            streams.close();
+        }
+    }
+
     @Test
     public void testStateChanges() throws InterruptedException {
         final StreamsBuilder builder = new StreamsBuilder();
         final KafkaStreams streams = new KafkaStreams(builder.build(), props);
 
-        final StateListenerStub stateListener = new StateListenerStub();
-        streams.setStateListener(stateListener);
-        Assert.assertEquals(streams.state(), KafkaStreams.State.CREATED);
-        Assert.assertEquals(stateListener.numChanges, 0);
+        try {
```

Comment:
If we reuse shared streams instance do we still need try-catch here as the `cleanup()` should close streams.